plugins/plugin-testing-java/plugin-testing-testng/che-plugin-testing-testng-server/src/main/java/org/eclipse/che/plugin/testing/testng/server/TestNGRunner.java

```diff
@@ -40,8 +71,114 @@ public TestResult execute(Map<String, String> testParameters) throws Exception {
     }
 
     @Override
+    @Nullable
     public ProcessHandler execute(TestExecutionContext context) {
+        String projectPath = context.getProjectPath();
+        IProject project = ResourcesPlugin.getWorkspace().getRoot().getProject(projectPath);
+        IJavaProject javaProject = JavaModelManager.getJavaModelManager().getJavaModel().getJavaProject(project);
+        if (javaProject.exists()) {
+            return startTestProcess(javaProject, context);
+        }
+
+        return null;
+    }
+
+    private ProcessHandler startTestProcess(IJavaProject javaProject, TestExecutionContext context) {
+        File suiteFile = createSuite(context, javaProject);
+        if (suiteFile == null) {
+            throw new RuntimeException("Can't create TestNG suite xml file.");
+        }
+
+        JavaParameters parameters = new JavaParameters();
+        parameters.setJavaExecutable("java");
+        parameters.setMainClassName("org.testng.CheTestNGLauncher");
+        parameters.setWorkingDirectory("/tmp");
```

Comment:
More precisely the `-d` command line option allows specifying where the reports will be generated (defaults to test-output).
See http://testng.org/doc/documentation-main.html#running-testng

So it seems to me there should be a solution to both:
- start the test process in the tested project root
- create the test outputs in a place that isn't visible in the explorer.